src/pybind/mgr/orchestrator.py

```diff
@@ -621,6 +621,14 @@ def update_rgw(self, spec):
         """
         raise NotImplementedError()
 
+    def get_container_image_id(self, image_name):
```

Comment:
That looks really low level to me. Is there a need for this api?

Reply:
I think so.  The upgrade check needs to know what the (latest) hash is for a given container image name.  We could put some generic code to query the registry directly into orchestrator_cli.py, perhaps...?